### ISSUE TITLE
Unexpected clusters with high z scores in ALE subtraction analysis

### ISSUE THREAD

**Issue by alexenge** [neurostuff/NiMARE#442]
## Summary
Whenever I run an ALE subtraction analysis using `meta.cbma.ALESubtraction()` with NiMARE v0.0.5, the resulting contrast map shows some unexpected clusters with very high _z_ scores (see output below). The size and shape of these clusters (with their very rough edges) suggests to me that these _z_ scores have not been estimated correctly.

## Additional details
- NiMARE version: 0.0.5

### What were you trying to do?

Running an ALE subtraction analysis between two data sets (i.e., two Sleuth files with non-overlapping sub-groups of experiments from a larger set of experiments).

### What did you expect to happen?

To get a subtraction map with _z_ scores showing the difference between `dset1` and `dset2`. Performing this analysis in GingerALE or NiMARE v0.0.3 shows some clusters with _z_ scores of up to ~4, in plausible locations and with a smooth shape at the edges.

### What actually happened?

Performing the same analysis in NiMARE v0.0.5 produces very similar results throughout most of the brain, but there are also some weird-looking additional clusters, all with a rough-edged shape and at implausible locations (e.g., in the cerebellum). Note that the same happens with different input data sets, with `low_memory=True`, and with higher a number of iterations (e.g., `n_iters=50000`).

### Input
[older.txt](https://github.com/neurostuff/NiMARE/files/5975454/older.txt)
[younger.txt](https://github.com/neurostuff/NiMARE/files/5975455/younger.txt)

### Code

```python
# Read Sleuth files
from nimare import io, meta
dset1 = io.convert_sleuth_to_dataset(text_file='older.txt')
dset2 = io.convert_sleuth_to_dataset(text_file='younger.txt')

# Perform subtraction analysis
sub = meta.cbma.ALESubtraction(n_iters=10000, low_memory=False)
sres = sub.fit(dset1, dset2)

# Save + plot
from nibabel import save
from nilearn import plotting
img_z = sres.get_map('z_desc-group1MinusGroup2')
save(img_z, filename='older_minus_younger.nii.gz')
p = plotting.plot_glass_brain('older_minus_younger.nii.gz', colorbar=True)
p.savefig('older_minus_younger.png')
```

### Output
[older_minus_younger.nii.gz](https://github.com/neurostuff/NiMARE/files/5976276/older_minus_younger.nii.gz)
![older_minus_younger](https://user-images.githubusercontent.com/66270814/107858661-8514ff80-6e35-11eb-8c73-23a5bb33afc0.png)

**Comment by tsalo** [comment]
Thanks for the bug report @alexenge. It looks like this is due to the changes we made in #245. I will try to dig into this tomorrow or early next week.

**Comment by tsalo** [comment]
@alexenge, I just want to let you know that we haven't forgotten about this issue, even if my initial timeline was a little optimistic (sorry about that!). The other devs have been discussing it and we'll try to follow up here in the next couple of days.

**Comment by alexenge** [comment]
Thanks a lot, Taylor and everyone, for looking deeper into this! No hurry and let me know if I can help.

**Comment by tyarkoni** [comment]
@alexenge this should be fixed by #447; the patch will hopefully hit master soon.  Here's what your code generates post-fix:

![older_minus_younger](https://user-images.githubusercontent.com/303932/109053487-5f43f200-76a2-11eb-9274-256a19217b1b.png)

Thanks for bringing this to our attention!

**Comment by alexenge** [comment]
@tyarkoni and @tsalo, I've just tried some subtraction analyses using the version on the `fix-perm-p-vals` branch and the results look very plausible to me. Thanks a lot for your efforts!
